crates/pairwise/tests/common/mod.rs

```diff
@@ -0,0 +1,108 @@
+// the reason this is named mod.rs has to do with some complexities of how
+// testing is handled
+//
+// we are following the advice of the rust book
+// https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests
```

Comment:
Trying to check my understanding: would this work if it were `tests/common.rs`?

Reply:
Because it is in the tests directory, it would not work. That's part of the weirdness that the comment acknowledges

I think the basic idea is that the compiler tries to to compile every file in the tests directory as a separate binary (filled with tests). The exception has to do with these `mod.rs` files in subdirectories. There is historical significance to files named `mod.rs`.